django-stubs/contrib/gis/geoip2.pyi

```diff
@@ -0,0 +1,75 @@
+from collections.abc import Sequence
+from ipaddress import IPv4Address, IPv6Address
+from typing import Final, Literal, TypedDict
+
+from _typeshed import StrPath
+from django.contrib.gis.geos import Point
+from django.utils.functional import cached_property
+from typing_extensions import Self, TypeAlias, deprecated
+
+HAS_GEOIP2: bool
+SUPPORTED_DATABASE_TYPES: set[str]
+
+_CoordType: TypeAlias = tuple[float, float] | tuple[None, None]
+_QueryType: TypeAlias = str | IPv4Address | IPv6Address
+
+class _CityResponse(TypedDict):
+    accuracy_radius: int | None
+    city: str
+    continent_code: str
+    continent_name: str
+    country_code: str
+    country_name: str
+    is_in_european_union: bool
+    latitude: float
+    longitude: float
+    metro_code: int | None
+    postal_code: str | None
+    region_code: str | None
+    region_name: str | None
+    time_zone: str | None
+    # Kept for backward compatibility.
+    dma_code: int | None
+    region: str | None
+
+class _CountryResponse(TypedDict):
+    continent_code: str
+    continent_name: str
+    country_code: str
+    country_name: str
+    is_in_european_union: bool
+
+class GeoIP2Exception(Exception): ...
+
+class GeoIP2:
+    MODE_AUTO: Final = 0
+    MODE_MMAP_EXT: Final = 1
+    MODE_MMAP: Final = 2
+    MODE_FILE: Final = 4
+    MODE_MEMORY: Final = 8
+    cache_options: Final[frozenset[Literal[0, 1, 2, 4, 8]]]
+
+    def __init__(
+        self,
+        path: StrPath | None = ...,
+        cache: Literal[0, 1, 2, 4, 8] = ...,
+        country: str | None = ...,
+        city: str | None = ...,
```

Comment:
```suggestion
        path: StrPath | None = None,
        cache: int = 0,
        country: str | None = None,
        city: str | None = None,
```

Reply:
It is hard to infer `Literal[0, 1, 2]`. For example, the simplest operations like `[0, 1][0]` or `cache_type = {"country": 0}; cache_type.get(name)` would infer `int` :(

Reply:
Ah, yeah. Of course.